src/mavsdk/plugins/camera/camera_impl.cpp

```diff
@@ -1557,17 +1560,34 @@ bool CameraImpl::should_fetch_camera_definition(const std::string& uri) const
            !_has_camera_definition_timed_out;
 }
 
+bool CameraImpl::starts_with(const std::string& str, const std::string& prefix)
+{
+    return str.compare(0, prefix.size(), prefix) == 0;
+}
+
 Camera::Result CameraImpl::fetch_camera_definition(
     const mavlink_camera_information_t& camera_information, std::string& camera_definition_out)
 {
-    auto result =
-        download_definition_file(camera_information.cam_definition_uri, camera_definition_out);
-
-    return result;
+    auto& uri = camera_information.cam_definition_uri;
+
+    if (starts_with(uri, "http://") || starts_with(uri, "https://")) {
+        LogInfo() << "Download file: " << uri << " using cURL...";
+        auto result = download_definition_file_curl(uri, camera_definition_out);
+        LogInfo() << "Downloaded file, result " << result;
+        return result;
+    } else if (starts_with(uri, "mftp://") || starts_with(uri, "mavlinkftp://")) {
```

Comment:
Is `mftp` a thing? I thought it was `mavftp` :grin:. But then I have never seen a drone exposing the camera definition over mavlink ftp, so... :grin:

Reply:
Yes, `mftp` is a thing:
https://github.com/mavlink/MAVSDK/blob/68c8041bc911e31f99c8575a042d6b3919583b39/src/mavsdk/plugins/component_metadata/component_metadata_test.cpp#L15

And so is `mavlinkftp` according to [CAMERA_INFORMATION](https://mavlink.io/en/messages/common.html#CAMERA_INFORMATION).

Not sure about `mavftp`, although that would be the nicest one...